crypto/hash/hmac_ossl.c

```diff
@@ -118,38 +205,69 @@ static srtp_err_status_t srtp_hmac_dealloc(srtp_auth_t *a)
 
 static srtp_err_status_t srtp_hmac_start(void *statev)
 {
-    HMAC_CTX *state = (HMAC_CTX *)statev;
-
-    if (HMAC_Init_ex(state, NULL, 0, NULL, NULL) == 0)
+    srtp_hmac_ossl_ctx_t *hmac = (srtp_hmac_ossl_ctx_t *)statev;
+
+#ifdef SRTP_OSSL_USE_EVP_MAC
+    if (hmac->use_dup) {
+        if (hmac->ctx != NULL) {
+            EVP_MAC_CTX_free(hmac->ctx);
+        }
+        hmac->ctx = EVP_MAC_CTX_dup(hmac->ctx_dup);
+        if (hmac->ctx == NULL) {
+            return srtp_err_status_alloc_fail;
+        }
+    } else {
+        if (EVP_MAC_init(hmac->ctx, NULL, 0, NULL) == 0) {
+            return srtp_err_status_auth_fail;
+        }
+    }
+#else
+    if (HMAC_Init_ex(hmac->ctx, NULL, 0, NULL, NULL) == 0)
         return srtp_err_status_auth_fail;
-
+#endif
     return srtp_err_status_ok;
 }
 
 static srtp_err_status_t srtp_hmac_init(void *statev,
                                         const uint8_t *key,
                                         int key_len)
 {
-    HMAC_CTX *state = (HMAC_CTX *)statev;
+    srtp_hmac_ossl_ctx_t *hmac = (srtp_hmac_ossl_ctx_t *)statev;
 
-    if (HMAC_Init_ex(state, key, key_len, EVP_sha1(), NULL) == 0)
-        return srtp_err_status_auth_fail;
+#ifdef SRTP_OSSL_USE_EVP_MAC
+    OSSL_PARAM params[2];
+
+    params[0] = OSSL_PARAM_construct_utf8_string("digest", "SHA1", 0);
+    params[1] = OSSL_PARAM_construct_end();
 
+    if (EVP_MAC_init(hmac->use_dup ? hmac->ctx_dup : hmac->ctx, key, key_len,
+                     params) == 0) {
```

Comment:
```suggestion
    if (EVP_MAC_init(hmac->ctx, key, key_len, params) == 0) {
```

Reply:
when using dup, doesn't the key need to be set on the dup_ctx ?  the normal ctx will be null at this point ??

Reply:
testing shows this to crash, so rejecting for now